### PR TITLE
Patch/window handling

### PR DESCRIPTION
**Pull request by nanoant**
This PR reduces number of references to PyQtGraph windows implicitly hold by the package. Also removes some memory leak in `ImageView.close()` not taking care of `self.imageItem`.

This PR is related to https://github.com/pyqtgraph/pyqtgraph/issues/466

### CODE REVIEW

File: pyqtgraph/__init__.py [hunk 1]
```diff
@@ -378,8 +378,6 @@ def exit():
 
 ## Convenience functions for command-line use
 
-plots = []
```
Comment (campagnola) [inline pyqtgraph/__init__.py]: These references should be necessary to ensure the plots stay alive as long as they the user keeps them open. Perhaps we can have the references automatically removed when the window is closed?

Comment (nanoant) [inline pyqtgraph/__init__.py]: Alright, I'll try to keep the references, and remove them on window close.

**Comment by nanoant** [comment]
@campagnola Hi I have updated my PR, but I have no idea why Travis fails with `IOError: [Errno 4] Interrupted system call`. I don't see any of updated code in the call-stack. Any clues?

**Comment by j9ac9k** [comment]
Hi @nanoant sorry it's taken us so long to get to this.  I fixed a merge conflict, this looks good to me.  I'll be merging.

**Pull request merged**